src/compute-types/src/dataflows.rs

```diff
@@ -357,6 +357,14 @@ impl<P, S, T> DataflowDescription<P, S, T> {
         assert!(builds.next().is_none());
         build
     }
+
+    /// Whether this dataflow is a materialized view.
+    pub fn is_materialized_view(&self) -> bool {
+        self.index_exports.is_empty()
+            && self.sink_exports.iter().all(|(_, sink_desc)| {
+                matches!(sink_desc.connection, ComputeSinkConnection::Persist(_))
+            })
+    }
```

Comment:
If we ever start exporting multiple objects from a dataflow, this code will break, so I think we should have an assert here that only a single export exists. But I'm also proposing not having the constraint in `create_dataflow` in the first place, so maybe we don't even need this method?

Reply:
Yes, deleting this now.